pkg/lock/lock_solaris.go

```diff
@@ -51,16 +42,24 @@ func LockedOpenFile(path string, flag int, perm os.FileMode) (*LockedFile, error
 		return nil, fmt.Errorf("Unsupported flag (%d)", flag)
 	}
 
-	lock.Type = lockType
-	lock.Whence = 0
+	var lock = syscall.Flock_t{
+		Start:  0,
+		Len:    0,
+		Pid:    0,
+		Type:   lockType,
+		Whence: 0,
```

Comment:
Yes it is indicate a meaning. That we are reading the right ranges and they outght to be these values.